File: USACO 2021 DEC Bronze/lonely-photo.cpp
#include </Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/bits/stdc++.h>

using namespace std;

int main()
{
    int n;
    string cows;
    cin >> n >> cows;
    int gCount = 0;
    int hCount = 0;
    string substring;
    for (int i = 0; i <= cows.length()-3; i++) {
        substring = cows.substr(i, 3);
        for (int j = 0; j < substring.length(); j++) {
            if (substring[j] == "H") {
                cout << "Found a H";
            }
            else {
                cout << "Found a G";
            }
        }
    }
    return 0;
}

File: USACO 2021 DEC Bronze/lonely-photo.py
n = int(input())
cows = input()
lonelyPhotos = 0
hCount = 0
gCount = 0
substringLength = 3
for i in range(n-(substringLength-1)):
    substring = cows[i:i+substringLength]
    for i in range(len(substring)):
        if substring[i] == 'H':
            hCount += 1
        else:
            gCount += 1
    if hCount == 1 or gCount == 1:
        lonelyPhotos += 1
    gCount = 0
    hCount = 0
print(lonelyPhotos)
